src/hardware/sblaster.cpp

```diff
@@ -1320,56 +1289,6 @@ static void flush_remaining_dma_transfer()
 	}
 }
 
-static void per_tick_callback();
-
-static void per_frame_callback(uint32_t);
-
-static void add_next_frame_callback()
-{
-	assert(sblaster);
-	assert(sblaster->channel);
-
-	PIC_AddEvent(per_frame_callback, sblaster->channel->GetMillisPerFrame());
-}
-
-void CallbackType::SetNone()
-{
-	if (timing_type != TimingType::None) {
-
-		(timing_type == TimingType::PerTick)
-		        ? TIMER_DelTickHandler(per_tick_callback)
-		        : PIC_RemoveEvents(per_frame_callback);
-
-		timing_type = TimingType::None;
-	}
-}
-
-void CallbackType::SetPerTick()
-{
-	if (timing_type != TimingType::PerTick) {
-
-		SetNone();
-
-		frames_added_this_tick = 0;
-
-		TIMER_AddTickHandler(per_tick_callback);
-
-		timing_type = TimingType::PerTick;
-	}
-}
-
-void CallbackType::SetPerFrame()
-{
-	if (timing_type != TimingType::PerFrame) {
-
-		SetNone();
-
-		add_next_frame_callback();
-
-		timing_type = TimingType::PerFrame;
-	}
-}
-
```

Comment:
Oh man, I love when these emulator-specific complexities go away 💯 The closer the emulation code is to what's actually happening on the hardware, the better 🚀 You usually end up with complex emulator code when you fudge things.